webapp/channels/src/components/invitation_modal/invite_view.tsx

```diff
@@ -133,42 +132,37 @@ export default function InviteView(props: Props) {
 
     const errorProperties = {
         showError: false,
-        errorMessageId: '',
-        errorMessageDefault: '',
+        errorMessage: messages.exceededMaxBatch,
         errorMessageValues: {
-            text: '',
+            text: Constants.MAX_ADD_MEMBERS_BATCH.toString(),
         },
-        extraErrorText: '',
     };
 
     if (props.usersEmails.length > Constants.MAX_ADD_MEMBERS_BATCH) {
         errorProperties.showError = true;
-        errorProperties.errorMessageId = t(
-            'invitation_modal.invite_members.exceeded_max_add_members_batch',
-        );
-        errorProperties.errorMessageDefault = 'No more than **{text}** people can be invited at once';
-        errorProperties.errorMessageValues.text = Constants.MAX_ADD_MEMBERS_BATCH.toString();
     }
 
-    let placeholder = formatMessage({
-        id: 'invite_modal.add_invites',
-        defaultMessage: 'Enter a name or email address',
-    });
-    let noMatchMessageId = t(
-        'invitation_modal.members.users_emails_input.no_user_found_matching',
-    );
-    let noMatchMessageDefault =
-        'No one found matching **{text}**. Enter their email to invite them.';
-
-    if (!props.emailInvitationsEnabled) {
+    let placeholder;
+    let noMatchMessage;
+    if (props.emailInvitationsEnabled) {
+        placeholder = formatMessage({
```

Comment:
Not sure why we are mixing having the `messages` object, and also using the identifiers directly here. 0/5

Reply:
We could make things more complicated and create a "valued message descriptor" that we would have to process where it is used to get the proper string and values. Something like `type VMessageDescriptor = MessageDescriptor & {values?: ValuesType}`. Then we will have something like `function renderValuedDescriptor(vd, intl) { return intl.formatMessage({id: vd.id, defaultMessage: vd.defaultMessage}, vd.values);}`.

But 0/5 on that approach. Maybe we can figure out something in the middle that make sense.

Reply:
There's a few places we have done that, but I'm not sure if I'd want to do that everywhere because it is nice to encourage people to not overuse values in cases where they should just have two separate strings. Values also don't always work when the message is being used for an aria-label or something else that needs text.

For now, I'll leave it -as-is, but I'm going to keep thinking about this since it would be nice if there was a good solution for this that worked everywhere